### PR TITLE
feat(schema 5.1.0): unique spatial assay ontology term id

### PR DESCRIPTION
**Pull request by MillenniumFalconMechanic**
## Reason for Change

- #873 

## Changes

- Added unique constraint to assay ontology term id for spatial assay.
- Refactored calls to `_validate_spatial_tissue_position(tissue_position_name...` into separate function for testability.

## Testing

- Added/updated unit tests.

### CODE REVIEW

File: cellxgene_schema_cli/tests/test_validate.py [hunk 1]
```diff
@@ -835,7 +859,7 @@ def test__validate_tissue_position_int_max_error(self, tissue_position_name, max
         "cell_type_ontology_term_id, in_tissue",
         [("unknown", 0), (["unknown", "CL:0000066"], [0, 1]), ("CL:0000066", 1)],
     )
-    def test_validate_cell_type_ontology_term_id_ok(self, cell_type_ontology_term_id, in_tissue):
+    def test__validate_cell_type_ontology_term_id_ok(self, cell_type_ontology_term_id, in_tissue):
```
Comment (MillenniumFalconMechanic) [inline cellxgene_schema_cli/tests/test_validate.py:862]: Standardized test name, here and below.

File: cellxgene_schema_cli/tests/test_validate.py [hunk 2]
```diff
@@ -770,11 +793,12 @@ def test__validate_tissue_position_required(self, tissue_position_name):
             f"obs['{tissue_position_name}'] {ERROR_SUFFIX_VISIUM_AND_IS_SINGLE_TRUE_REQUIRED}." in validator.errors[0]
         )
 
-    def test__validate_tissue_position_not_required(self):
```
Comment (MillenniumFalconMechanic) [inline cellxgene_schema_cli/tests/test_validate.py:773]: Moved to parameterized to avoid `["EFO:0010961", "EFO:0030062"]` triggering unique error.

**Review by nayib-jose-gloria — APPROVED** [review]

**Review by Bento007 — APPROVED** [review]

**Comment by MillenniumFalconMechanic** [comment]
Hi @brian-mott, this is ready for review!

File: cellxgene_schema_cli/cellxgene_schema/validate.py [hunk 3]
```diff
@@ -1501,6 +1502,29 @@ def _validate_spatial_is_primary_data(self):
...
+
+        # Identical check is only applicable to spatial datasets.
+        if not self._is_supported_spatial_assay():
+            return
+
+        # Validate assay ontology term ids are identical.
+        term_count = obs["assay_ontology_term_id"].nunique()
+        if term_count > 1:
+            self.errors.append(
+                "When obs['assay_ontology_term_id'] is either 'EFO:0010961' (Visium Spatial Gene Expression) or "
```
Comment (brian-mott) [inline cellxgene_schema_cli/cellxgene_schema/validate.py:1524]: If I add 'EFO:0009922' (10x 3' v3) as an `assay_ontology_term_id` for one obs in a spatial dataset, the validation fails but I don't get this error or any errors

**Review by brian-mott — APPROVED** [review]
Looks good with spatial and non spatial datasets and if is_single is false. Sorry for the method mishap, thank you for that catch!

**Pull request merged**